exchanges/lbank/lbank.go

```diff
@@ -0,0 +1,583 @@
+		lbankCurrencyPairs)
+
+	var result []string
+	return result, l.SendHTTPRequest(path, &result)
+}
+
+// GetMarketDepths returns arrays of asks, bids and timestamp
+func (l *Lbank) GetMarketDepths(symbol, size, merge string) (MarketDepthResponse, error) {
+	var m MarketDepthResponse
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	params.Set("size", size)
+	params.Set("merge", merge)
+	path := fmt.Sprintf("%s/v%s/%s?%s", l.APIUrl, lbankAPIVersion, lbankMarketDepths, params.Encode())
+	return m, l.SendHTTPRequest(path, &m)
+}
+
+// GetTrades returns an array of available trades regarding a particular exchange
+func (l *Lbank) GetTrades(symbol, size, time string) ([]TradeResponse, error) {
+	var g []TradeResponse
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	params.Set("size", size)
+	params.Set("time", time)
+	path := fmt.Sprintf("%s/v%s/%s?%s", l.APIUrl, lbankAPIVersion, lbankTrades, params.Encode())
+	return g, l.SendHTTPRequest(path, &g)
+}
+
+// GetKlines returns kline data
+func (l *Lbank) GetKlines(symbol, size, klineType, time string) ([]KlineResponse, error) {
+	var klineTemp interface{}
+	var k []KlineResponse
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	params.Set("size", size)
+	params.Set("type", klineType)
+	params.Set("time", time)
+	path := fmt.Sprintf("%s/v%s/%s?%s", l.APIUrl, lbankAPIVersion, lbankKlines, params.Encode())
+	err := l.SendHTTPRequest(path, &klineTemp)
+	if err != nil {
+		return k, err
+	}
+
+	resp, ok := klineTemp.([]interface{})
+	if !ok {
+		return nil, errors.New("response received is invalid")
+	}
+
+	for i := range resp {
+		resp2, ok := resp[i].([]interface{})
+		if !ok {
+			return nil, errors.New("response received is invalid")
+		}
+		var tempResp KlineResponse
+		for x := range resp2 {
+			switch x {
+			case 0:
+				tempResp.TimeStamp = int64(resp2[x].(float64))
+			case 1:
+				if val, ok := resp2[x].(int64); ok {
+					tempResp.OpenPrice = float64(val)
+				} else {
+					tempResp.OpenPrice = resp2[x].(float64)
+				}
+			case 2:
+				if val, ok := resp2[x].(int64); ok {
+					tempResp.HigestPrice = float64(val)
+				} else {
+					tempResp.HigestPrice = resp2[x].(float64)
+				}
+			case 3:
+				if val, ok := resp2[x].(int64); ok {
+					tempResp.ClosePrice = float64(val)
+				} else {
+					tempResp.ClosePrice = resp2[x].(float64)
+				}
+			case 4:
+				if val, ok := resp2[x].(int64); ok {
+					tempResp.TradingVolume = float64(val)
+				} else {
+					tempResp.TradingVolume = resp2[x].(float64)
+				}
+			}
+		}
+		k = append(k, tempResp)
+	}
+	return k, nil
+}
+
+// GetUserInfo gets users account info
+func (l *Lbank) GetUserInfo() (InfoResponse, error) {
+	var resp InfoResponse
+	path := fmt.Sprintf("%s/v%s/%s?", l.APIUrl, lbankAPIVersion, lbankUserInfo)
+	err := l.SendAuthHTTPRequest("POST", path, nil, &resp)
+	if err != nil {
+		return resp, err
+	}
+
+	if resp.Error != 0 {
+		return resp, ErrorCapture(resp.Error)
+	}
+
+	return resp, nil
```

Comment:
Always returns empty data due to invalid unmarshaling 
(See feedback on InfoResponse struct)